src/validators/mod.rs

```diff
@@ -83,11 +84,13 @@ impl SchemaValidator {
             Some(t) => t.into_py(py),
             None => validator.get_name().into_py(py),
         };
+        let frozen: bool = schema_or_config_same(schema.downcast()?, config, intern!(py, "frozen"))?.unwrap_or(false);
```

Comment:
frozen can't be set on the model schema, so I think you just want

```rs
let frozen: bool =config.get_as(intern!(py, "frozen"))?.unwrap_or(false);
```
Or something.

Also, you can inline at line 93.

Reply:
I was thinking if were to set it on the model schema we could replace `FrozenSetSchem` with `SetSchema` but with a frozen to true. We could use it the same way for other collection types as well.